libarchive/entry.py

```diff
@@ -61,30 +61,53 @@ def isfifo(self):
 
     @property
     def islnk(self):
+        return bool(ffi.entry_hardlink_w(self._entry_p))
+
+    @property
+    def issym(self):
         return self.filetype & 0o170000 == 0o120000
 
+    def _linkpath(self):
+        return (ffi.entry_symlink(self._entry_p)
+                or ffi.entry_hardlink(self._entry_p))
+
+    # aliases to get the same api as tarfile
+    linkpath = property(_linkpath)
+    linkname = property(_linkpath)
+
     @property
     def isreg(self):
         return self.filetype & 0o170000 == 0o100000
 
     @property
+    def isfile(self):
+        return self.isreg
+
+    @property
     def issock(self):
         return self.filetype & 0o170000 == 0o140000
 
     @property
+    def isdev(self):
+        return self.ischr or self.isblk or self.isfifo or self.issock
+
+    @property
     def mtime(self):
         return ffi.entry_mtime(self._entry_p)
 
-    @property
-    def pathname(self):
-        return ffi.entry_pathname_w(self._entry_p)
+    def _getpathname(self):
+        return ffi.entry_pathname(self._entry_p)
```

Comment:
I think what we want is `return ffi.entry_pathname_w(self._entry_p) or ffi.entry_pathname(self._entry_p)`.